plugins/module_utils/common/argspec_validate.py

```diff
@@ -0,0 +1,212 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""Use AnsibleModule's argspec validation
+
+def _check_argspec(self):
+    aav = AnsibleArgSpecValidator(
+        data=self._task.args,
+        schema=DOCUMENTATION,
+        schema_format="doc",
+        schema_conditionals={},
+        other_args={},
+        name=self._task.action,
+    )
+    valid, errors = aav.validate()
+    if not valid:
+        raise AnsibleActionFail(errors)
+
+"""
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import json
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.ansible.utils.plugins.module_utils.common.utils import (
+    dict_merge,
+)
+from ansible.module_utils.six import iteritems, string_types
+from ansible.module_utils._text import to_bytes
+
+try:
+    import yaml
+
+    try:
+        from yaml import CSafeLoader as SafeLoader
+    except ImportError:
+        from yaml import SafeLoader
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+try:
```

Comment:
Can you please add the intention of having this import maybe as a `TODO:` section

Reply:
yep, good idea